tests/rptest/tests/maintenance_test.py

```diff
@@ -185,7 +185,31 @@ def _verify_cluster(self, target, target_expect):
 
     def _maintenance_disable(self, node):
         if self._use_rpk:
-            self.rpk.cluster_maintenance_disable(node)
+            """
+            "Requested node does not exists" is logged as a warning from admin_server::log_exception.
+                if (ex.status() == http_status::internal_server_error) {
+                    vlog(logger.error, "{}", log_ex(status));
+                } else if (status >= 400) {
+                    vlog(logger.warn, "{}", log_ex(status)); // <-- this branch
+                }
+
+            This appears to occur because the health monitor failed to get node health report for exponential_backoff seconds.
+            Example: health_monitor_backend.cc:540 - unable to get node health report from 2 - rpc::errc::exponential_backoff, marking node as down
+            
+            Since this is caused by a warning with a backoff, we retry the request.
+            """
+            def do_rpk_maint_disable():
+                try:
+                    self.rpk.cluster_maintenance_disable(node)
+                    return True
+                except RpkException as e:
+                    assert self._use_rpk
+                    e_as_str = str(e)
+                    if "Requested node does not exists" in e_as_str and "503" in e_as_str:
+                        return False
+                    raise e
+
+            wait_until(do_rpk_maint_disable, timeout_sec=30, backoff_sec=1)
```

Comment:
Here is an example of the full log:
```
INFO  2023-05-03 14:51:42,434 [shard 0] raft - [group_id:0, {redpanda/controller/0}] vote_stm.cc:279 - became the leader term: 2
INFO  2023-05-03 14:51:46,823 [shard 0] cluster - health_monitor_backend.cc:554 - received node 1 health report, marking node as up
INFO  2023-05-03 14:51:46,829 [shard 0] cluster - health_monitor_backend.cc:554 - received node 2 health report, marking node as up
WARN  2023-05-03 14:51:56,831 [shard 0] cluster - health_monitor_backend.cc:540 - unable to get node health report from 1 - rpc::errc::exponential_backoff, marking node as down
INFO  2023-05-03 14:51:57,348 [shard 1] r/heartbeat - heartbeat_manager.cc:209 - Closed unresponsive connection to 1
INFO  2023-05-03 14:51:57,366 [shard 0] r/heartbeat - heartbeat_manager.cc:209 - Closed unresponsive connection to 1
INFO  2023-05-03 14:51:57,512 [shard 1] r/heartbeat - heartbeat_manager.cc:209 - Closed unresponsive connection to 1
INFO  2023-05-03 14:51:57,534 [shard 0] r/heartbeat - heartbeat_manager.cc:209 - Closed unresponsive connection to 1
WARN  2023-05-03 14:51:58,328 [shard 0] admin_api_server - admin_server.cc:502 - [_anonymous] exception intercepted - url: [http://docker-rp-6:9644/v1/brokers/1/maintenance] http_return_status[503] reason - seastar::httpd::base_exception (Unexpected error: Requested node does not exists)
WARN  2023-05-03 14:51:59,925 [shard 0] admin_api_server - admin_server.cc:502 - [_anonymous] exception intercepted - url: [http://docker-rp-6:9644/v1/brokers/1/maintenance] http_return_status[503] reason - seastar::httpd::base_exception (Unexpected error: Requested node does not exists)
WARN  2023-05-03 14:52:01,481 [shard 0] admin_api_server - admin_server.cc:502 - [_anonymous] exception intercepted - url: [http://docker-rp-6:9644/v1/brokers/1/maintenance] http_return_status[503] reason - seastar::httpd::base_exception (Unexpected error: Requested node does not exists)
```

Reply:
https://redpandadata.slack.com/archives/C04B5T58QQ1/p1683315680236749?thread_ts=1683307800.545609&cid=C04B5T58QQ1

Reply:
> https://redpandadata.slack.com/archives/C04B5T58QQ1/p1683315680236749?thread_ts=1683307800.545609&cid=C04B5T58QQ1

To explain what @bharathv said in the thread:
We should not need to check the drain status because the status is served from a cache and `set_maintenance_mode` is idempotent. Therefore, it does not hurt to issue the command without those checks.

(Referring to the error logs): Furthermore, the maintenance mode command should work async. So when the node comes back up and it catches up with the controller log, the node should apply the maintenance mode command.

Therefore, the changes I made in `1c9412a` simply remove the drain status check from RPK and force any maintenance disable request to go to the controller leader. I also removed the checks I put into the Admin server on the RP side.